Adding queryAll operation to Bulk

petergtam opened the pull request:
Adding queryAll operation to Bulk

Totorokrut commented:
That is what I was finding actually! :) Thank you.

besteman commented:
Can we merge this?

besteman commented:
Can we merge this?
@andscoop
@jgillmanjr

andscoop commented:
We'd really like to see testing for any feature additions or changes at this stage in the libs lifecycle. Can you add tests? @petergtam @besteman

petergtam commented:
Thanks @andscoop, working on it

nirgal commented on README.rst:
```diff
@@ -274,6 +274,14 @@ Query records:
 
     sf.bulk.Account.query(query)
 
+Query all records:
```
Please explain what this means (deleted records) in the README. This will be confusing.

dmartindelorme commented:
Hello, any news on this? I'd really like to use that feature cc @petergtam.

petergtam commented:
Hi everyone, I added the tests missing for it to be merge. The tests should also work to merge #280 . @dmartindelorme , @andscoop , @nickcatal , @besteman. Also I added a description to queryAll as suggested by @nirgal.

dmartindelorme commented:
Thanks for the reactivity!

tseader commented:
So what needs to happen to see this merged?  Would love to have this available b/c it's a big problem right now to not be able to get everything with bulk queries.

petergtam commented:
Hi everyone, @MatteoF1 added a really nice feature to my branch!

petergtam commented on simple_salesforce/bulk.py:
```diff
@@ -90,7 +90,7 @@ def _create_job(self, operation, object_name, use_serial,
             use_serial = 0
```
Do you guys know what is the reason for this? According to the docs it is expecting `Parallel` or `Serial`

jon-wobken replied:
It also allows for 0/1 in the concurrency mode header but its just not stated in the documents. If you pass 0 in concurrency mode then it will run the batches in parallel if you pass 1 then it will run them in serial.

petergtam replied:
Thanks for the clarification @jon-wobken

petergtam commented on simple_salesforce/tests/test_bulk.py:
```diff
@@ -0,0 +1,565 @@
+                            instance_url=tests.SERVER_URL,
+                            session=session)
+        contact = client.bulk.Contact.insert(data)
+        self.assertEqual([self.expected], contact)
```
I added `[self.expected]` to insert,update, upsert and so on because of https://github.com/simple-salesforce/simple-salesforce/pull/259/files#diff-0f8e08704aa000282b5b1703b2c78af3L226. I wonder if it would be better to use `extend`.

KourosNL commented:
Any updates on when this feature will be merged?

petergtam commented:
I'd like to know as well! Maybe @jon-wobken could help us?

jon-wobken commented:
@KourosNL @petergtam - Haven't had a chance to get back into this with a few other projects. Will look to take a quick pass reviewing this before the end of the week and then merge but quick glance at it looks good.

dspohn-adwerx commented:
I'm very much looking forward to this functionality, it will be great. Thank you so much for working on this!

EtherRx commented:
Desperate for this!  @jon-wobken Hoping you can find some time to make the last changes and publish a new build.  Happy to buy a coffee or donate to a charity of your choice!

shenry10 commented:
Cannot wait for this functionality!  This will be very important for my company.  @jon-wobken thank you for all the hard work, I am looking forward to the new build being published (and I am also very willing to donate to a charity of your choice along with @EtherRx).
